linera-views/src/backends/lru_caching.rs

```diff
@@ -50,54 +58,267 @@ static NUM_CACHE_SUCCESS: LazyLock<IntCounterVec> = LazyLock::new(|| {
 /// keep track of this.
 
 /// The data structures
+#[allow(clippy::type_complexity)]
 struct LruPrefixCache {
-    map: BTreeMap<Vec<u8>, Option<Vec<u8>>>,
-    queue: LinkedHashMap<Vec<u8>, (), RandomState>,
+    map_find_key_values: BTreeMap<Vec<u8>, Vec<(Vec<u8>, Vec<u8>)>>,
+    map_find_keys: BTreeMap<Vec<u8>, Vec<Vec<u8>>>,
+    map_contains_key: BTreeMap<Vec<u8>, bool>,
+    map_read_value: BTreeMap<Vec<u8>, Option<Vec<u8>>>,
```

Comment:
Would it be simpler to have a single `BTreeMap<Vec<u8>, CacheEntry>` with:

```rust
enum CacheEntry {
    DoesNotExist,
    Exists,
    Value(Vec<u8>),
}
```

`read_value` would automatically store a `Value` in there, but that `Value` entry can also count as `Exists` in a `contains_key` call.

Currently, each `map_read_value` entry makes the corresponding `map_contains_key` entry redundant.

Reply:
This approach has been taken and yes it leads to simplification of the code.